src/frontend/src/binder/relation/mod.rs

```diff
@@ -82,25 +82,26 @@ impl Relation {
 
     pub fn collect_correlated_indices_by_depth_and_assign_id(
         &mut self,
+        depth: Depth,
         correlated_id: CorrelatedId,
     ) -> Vec<usize> {
         match self {
             Relation::Subquery(subquery) => subquery
```

Comment:
Just, double-checking, does this also detect correlated input ref in FRO subquery?

(I think we do not currently support subquery decorrelation for FROM subquery, but still we want to detect correlated)

Reply:
Oh yes, indeed, this change explicity enables such detection and correct depth estimate.

Reply:
Yes, it could detect `correlated_input_ref` in FROM subquery. We already implement `ApplyJoinRule` which is used to handle the correlated ref under the join which belongs to `correlated input ref in FROM subquery`.